Add `Content-Type: application/json` for fiber requests

afbora opened the pull request:
### Fixes

- Fixes blocking requests by ModSecurity (owasp rules)

### Breaking changes
None

## Ready?
<!--
If you can help to check off the following tasks, that'd be great.
If not, don't worry - we will take care of it.

More details: https://contribute.getkirby.com
-->

- [x] Unit tests for fixed bug/feature
- [x] In-code documentation (wherever needed)
- [x] Tests and checks all pass


### For review team
<!-- 
We will take care of the following before merging the PR.
-->

- [x] Add changes to release notes draft in Notion
- [x] Add to [website docs release checklist](https://github.com/getkirby/getkirby.com/pulls) (if needed)

afbora committed:
Add `Content-Type: application/json` for fiber requests

afbora commented on panel/src/fiber/index.js:
```diff
@@ -235,6 +235,7 @@ export default class Fiber {
 				cache: "no-store",
 				headers: {
 					...this.options.headers(),
+					"Content-Type": "application/json",
```
@lukasbestle Does it matter if it is lowercase or not?

afbora replied:
I just wonder because it's lowercased in `api::request()`.

lukasbestle replied:
[MDN says](https://developer.mozilla.org/en-US/docs/Web/API/Headers) that `fetch` handles headers case-insensitively, so I guess it doesn't matter. The same applies to the HTTP spec itself.

afbora replied:
ActualIy I mean for overwriting content type.

lukasbestle replied:
I just tested it to make sure:

```js
fetch('/', {
  headers: {
    "Content-Type": "application/json",
    "content-type": "text/something"
  }
});
```

In Safari 15.6.1, `fetch()` merges the two unfortunately, so the request header becomes `application/json, text/something`.

That's indeed not great as it means that the headers are treated *partly* case-sensitively.

afbora replied:
What do you suggest? 

- Always use lowercase?
- Is it always to lowercase the keys before merging them? 

Do you have another idea?

lukasbestle replied:
> Is it always to lowercase the keys before merging them?

I think that's a great idea!

distantnative replied:
We should do it then in `request.js` as well

afbora replied:
Done ✅

afbora committed:
Use always lowercased request headers

afbora commented on panel/src/api/request.js:
```diff
@@ -23,7 +25,7 @@ export default (config) => {
 				headers: {
 					"x-requested-with": "xmlhttprequest",
```
@distantnative Do we need `x-requested-with` for `fiber.js` too?

bastianallgeier replied:
We used this initially to detect fiber requests, but as far as I know, it's no longer needed.

afbora replied:
👍

bastianallgeier reviewed (dismissed):
Looks good to me. Do you want to remove the x-requested-with header or do we keep it?

afbora committed:
Remove `x-requested-with` header

We used this initially to detect fiber requests, but as far as I know, it's no longer needed.
